xrpl/utils/txn_parser/get_balance_changes.py

```diff
@@ -0,0 +1,75 @@
+"""Parse balance changes of every account involved in the given transaction."""
+
+from decimal import Decimal
+from typing import List, Optional, Union
+
+from pydash import flatten  # type: ignore
+
+from xrpl.models import TransactionMetadata
+from xrpl.utils.txn_parser.utils import (
+    BalanceChanges,
+    get_quantities,
+    group_by_account,
+    normalize_nodes,
+)
+from xrpl.utils.txn_parser.utils.nodes import NormalizedNode
+from xrpl.utils.txn_parser.utils.types import Balance
+
+
+def _get_value(balance: Union[Balance, str]) -> Decimal:
+    if isinstance(balance, str):
+        return Decimal(balance)
+    return Decimal(balance["value"])
+
+
+def _compute_balance_change(node: NormalizedNode) -> Optional[Decimal]:
+    """
+    Get the balance change from a node.
+
+    Args:
+        node: The affected node.
+
+    Returns:
+        The balance change.
+    """
+    value: Optional[Decimal] = None
+    if node.get("NewFields") is not None:
+        assert node["NewFields"] is not None
+        if node["NewFields"].get("Balance"):
```

Comment:
In response to your comment above in the conversation.

As far as I understand, the issue you're running into here is that while `.get` does prove the entity exists, mypy doesn't know that `node.get("NewFields")` AND it finding a value is the same as `node["NewFields"]`.

I think you can avoid the assert if you do something like:
```
newFields = node.get("NewFields")
if(newFields is not None):
    if(newFields.get("Balance")):
...
```

That way mypy knows you've just proven that `NewFields` can't be `None`

Reply:
This is way better :)